projects/wave-core-new/src/lib/operators/dialogs/expression-operator/expression-operator.component.ts

```diff
@@ -0,0 +1,277 @@
+import {map, mergeMap} from 'rxjs/operators';
+import {Observable, Subscription, zip} from 'rxjs';
+
+import {AfterViewInit, ChangeDetectionStrategy, Component, OnDestroy} from '@angular/core';
+import {FormControl, FormGroup, ValidationErrors, Validators} from '@angular/forms';
+import {ResultTypes} from '../../result-type.model';
+import {DataType, DataTypes} from '../../datatype.model';
+import {Unit} from '../../unit.model';
+// import {LetterNumberConverter} from '../helpers/multi-layer-selection/multi-layer-selection.component';
+import {RasterLayer} from '../../../layers/layer.model';
+import {MappingRasterSymbology} from '../../../layers/symbology/symbology.model';
+import {WaveValidators} from '../../../util/form.validators';
+import {ProjectService} from '../../../project/project.service';
+
+/**
+ * This dialog allows calculations on (one or more) raster layers.
+ */
+@Component({
+    selector: 'wave-expression-operator',
+    templateUrl: './expression-operator.component.html',
+    styleUrls: ['./expression-operator.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class ExpressionOperatorComponent implements AfterViewInit, OnDestroy {
+    private static readonly RASTER_VALUE = 'value';
+    readonly RASTER_TYPE = [ResultTypes.RASTER];
+    readonly UNITLESS_UNIT = Unit.defaultUnit;
+    readonly CUSTOM_UNIT_ID = 'CUSTOM';
+
+    form: FormGroup;
+
+    outputDataTypes$: Observable<Array<[DataType, string]>>;
+    // outputUnits$: Observable<Array<Unit>>;
+    // outputUnitIsCustom$: Observable<boolean>;
```

Comment:
I guess you should create a lot of TODOs about the missing parts that you cannot implement right now.

Reply:
cb5f2338266a2226d1ba03174a13ae87a16e800a